src/lib/protocols/raft.c

```diff
@@ -0,0 +1,111 @@
+/*
+ * raft.c
+ *
+ * Copyright (C) 2024 - ntop.org
+ *
+ * This file is part of nDPI, an open source deep packet inspection
+ * library based on the OpenDPI and PACE technology by ipoque GmbH
+ *
+ * nDPI is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * nDPI is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with nDPI.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#include "ndpi_protocol_ids.h"
+
+#define NDPI_CURRENT_PROTO NDPI_PROTOCOL_RAFT
+#include "ndpi_api.h"
+#include "ndpi_private.h"
+
+PACK_ON
+struct raft_header {
+  uint64_t msg_type;
+  uint64_t msg_length;
+} PACK_OFF;
```

Comment:
I think that sooner or later we need to take a closer look if/when all these PACK attributes are really needed...

Reply:
I think for that specific case, it should be fine to remove `PACK_*` iff we assume that structs in C are always 4-byte aligned.
My fear is that different CCs may have a different struct alignment.